Src/DependencyCollector/Shared/Implementation/ProfilerSqlProcessingBase.cs

```diff
@@ -402,22 +413,29 @@ private void OnEndInternal(object exceptionObj, object thisObj)
             {
                 this.TelemetryTable.Remove(thisObj);
 
-                var exception = exceptionObj as Exception;
-                if (exception != null)
+                if (processTelemetryItem)
                 {
-                    telemetry.Success = false;
-                    telemetry.Properties.Add("ErrorMessage", exception.Message);
+                    var exception = exceptionObj as Exception;
+                    if (exception != null)
+                    {
+                        telemetry.Success = false;
+                        telemetry.Properties.Add("ErrorMessage", exception.Message);
+
+                        var sqlEx = exception as SqlException;
+                        telemetry.ResultCode = sqlEx != null ? sqlEx.Number.ToString(CultureInfo.InvariantCulture) : "0";
+                    }
+                    else
+                    {
+                        telemetry.Success = true;
+                        telemetry.ResultCode = "0";
```

Comment:
this is not required as `ResultCode` is optional. It's not relevant to this review though...